src/main/java/uk/gov/hmcts/reform/bailcaseapi/domain/handlers/presubmit/CurrentCaseStateUpdater.java

```diff
@@ -54,11 +50,18 @@ public PreSubmitCallbackResponse<BailCase> handle(
         if (bailCase.read(RECORD_DECISION_TYPE, String.class).orElse("")
             .equals(DecisionType.CONDITIONAL_GRANT.toString()) && (callback.getEvent() == Event.RECORD_THE_DECISION)) {
             currentCaseState = DECISION_CONDITIONAL_BAIL.toString();
+        } else if (callback.getEvent().equals(Event.RECORD_THE_DECISION)) {
```

Comment:
In this first if-else block first assignment will always be overwritten by the second one.